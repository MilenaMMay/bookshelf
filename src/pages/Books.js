import React from 'react';
import PropTypes from 'prop-types';
import PageTitle from '../components/PageTitle';
import ControlHeader from '../components/ControlHeader';
import List from '../components/List';
import TextContent from '../components/TextContent';

class Books extends React.Component {
  render() {
    return (
      <div>
        <PageTitle>Bookshelf</PageTitle>
        <ControlHeader addUrl={this.props.addBookUrl} />
        { this.props.books && this.props.books.length > 0 ?
          <List
            items={this.props.books}
            deleteHandler={this.props.deleteHandler}
          />
          :
          <TextContent>Please add a book to start!</TextContent>
        }
      </div>
    );
  }
}

Books.propTypes = {
  books: PropTypes.array,
  addBookUrl: PropTypes.string,
  deleteHandler: PropTypes.func
};

export default Books

