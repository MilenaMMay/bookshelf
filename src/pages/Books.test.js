import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import Books from './Books';

configure({ adapter: new Adapter() });

describe('Books', () => {
  it('renders empty content as specified in the snapshot', () => {
    const component = renderer.create(
      <Books
        books={null}
        addBookUrl='/'
        deleteHandler={()=>{}}
      />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders books as specified in the snapshot', () => {
    const component = renderer.create(
      <Books
        books={[
          {
            id: 10,
            url: '/book/10',
            title: 'Ten'
          },
          {
            id: 20,
            url: '/book/20',
            title: 'Twenty'
          }
        ]}
        addBookUrl='/'
        deleteHandler={()=>{}}
      />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
