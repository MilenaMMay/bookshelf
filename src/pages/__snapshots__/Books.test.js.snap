// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Books renders books as specified in the snapshot 1`] = `
<div>
  <h1
    className="sc-bdVaJa cQWkDZ"
  >
    Bookshelf
  </h1>
  <div>
    <a
      href="/"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-plus-circle fa-w-16 fa-2x "
        data-icon="plus-circle"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
    <hr />
  </div>
  <table>
    <tbody>
      <tr>
        <td>
          <a
            href="/book/10"
          >
            Ten
          </a>
        </td>
        <td>
          <div />
        </td>
        <td>
          <a
            href="#"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-trash-alt fa-w-14 fa-xs "
              data-icon="trash-alt"
              data-prefix="fas"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 84V56c0-13.3 10.7-24 24-24h112l9.4-18.7c4-8.2 12.3-13.3 21.4-13.3h114.3c9.1 0 17.4 5.1 21.5 13.3L312 32h112c13.3 0 24 10.7 24 24v28c0 6.6-5.4 12-12 12H12C5.4 96 0 90.6 0 84zm416 56v324c0 26.5-21.5 48-48 48H80c-26.5 0-48-21.5-48-48V140c0-6.6 5.4-12 12-12h360c6.6 0 12 5.4 12 12zm-272 68c0-8.8-7.2-16-16-16s-16 7.2-16 16v224c0 8.8 7.2 16 16 16s16-7.2 16-16V208zm96 0c0-8.8-7.2-16-16-16s-16 7.2-16 16v224c0 8.8 7.2 16 16 16s16-7.2 16-16V208zm96 0c0-8.8-7.2-16-16-16s-16 7.2-16 16v224c0 8.8 7.2 16 16 16s16-7.2 16-16V208z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <a
            href="/book/20"
          >
            Twenty
          </a>
        </td>
        <td>
          <div />
        </td>
        <td>
          <a
            href="#"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-trash-alt fa-w-14 fa-xs "
              data-icon="trash-alt"
              data-prefix="fas"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 84V56c0-13.3 10.7-24 24-24h112l9.4-18.7c4-8.2 12.3-13.3 21.4-13.3h114.3c9.1 0 17.4 5.1 21.5 13.3L312 32h112c13.3 0 24 10.7 24 24v28c0 6.6-5.4 12-12 12H12C5.4 96 0 90.6 0 84zm416 56v324c0 26.5-21.5 48-48 48H80c-26.5 0-48-21.5-48-48V140c0-6.6 5.4-12 12-12h360c6.6 0 12 5.4 12 12zm-272 68c0-8.8-7.2-16-16-16s-16 7.2-16 16v224c0 8.8 7.2 16 16 16s16-7.2 16-16V208zm96 0c0-8.8-7.2-16-16-16s-16 7.2-16 16v224c0 8.8 7.2 16 16 16s16-7.2 16-16V208zm96 0c0-8.8-7.2-16-16-16s-16 7.2-16 16v224c0 8.8 7.2 16 16 16s16-7.2 16-16V208z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Books renders empty content as specified in the snapshot 1`] = `
<div>
  <h1
    className="sc-bdVaJa cQWkDZ"
  >
    Bookshelf
  </h1>
  <div>
    <a
      href="/"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-plus-circle fa-w-16 fa-2x "
        data-icon="plus-circle"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
    <hr />
  </div>
  <p
    className="sc-bwzfXH kDFfNT"
  >
    Please add a book to start!
  </p>
</div>
`;
